### PR TITLE
bump(sys-block/open-iscsi): sync with upstream

### PR DESCRIPTION
**Pull request by mischief**
WIP, please don't merge yet. feedback welcome, let's get this right the first time :-)

### CODE REVIEW

**Comment by darrensessions** [comment]
will multipathd be included?

**Comment by marineam** [comment]
Well, for starters please just include a single ebuild.

**Comment by mischief** [comment]
@marineam would you prefer i prune ebuilds/files in the initial commit? i figured i would sync upstream and prune stuff in subsequent commits.

**Comment by mischief** [comment]
i put this pr here so we can get feedback. the other pr #673 seems dead.

**Comment by marineam** [comment]
Pruning now is fine, for history it'll be more interesting to see how we modify the ebuild so unused extras aren't that interesting.

I presume that this doesn't have any systemd units right?

**Comment by mischief** [comment]
@marineam there's no systemd units upstream.

**Comment by mischief** [comment]
@marineam ptal

**Comment by mischief** [comment]
this PR depends on coreos/portage-stable#265

File: sys-block/open-iscsi/files/iscsidev.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#ID_TYPE=disk
+#ID_PART_TABLE_TYPE=gpt
+#ID_VENDOR=IET
+#ID_SERIAL_SHORT=IET_00010001
+#SEQNUM=25775
+
+# do the removal
+if [[ $ACTION = 'remove' ]]; then
+  # nohup needed so this isn't constantly run...
+  nohup find -L /dev/disk/by-path/ -type l -lname ${DEVNAME} -exec rm "{}" + 2>/dev/null &
```
Comment (marineam) [inline sys-block/open-iscsi/files/iscsidev.sh]: Oh god, I hate this script and everything it stands for. This should really be implemented something that *just* sets gives udev env vars, run as `IMPORT{program}` and then leave managing symlinks to udev.

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	if kernel_is -lt 2 6 16; then
+		die "Sorry, your kernel must be 2.6.16-rc5 or newer!"
+	fi
+
+	# Needs to be done, as iscsid currently only starts, when having the iSCSI
+	# support loaded as module. Kernel builtion options don't work. See this for
+	# more information:
+	# https://groups.google.com/group/open-iscsi/browse_thread/thread/cc10498655b40507/fd6a4ba0c8e91966
+	# If there's a new release, check whether this is still valid!
+	CONFIG_CHECK_MODULES="SCSI_ISCSI_ATTRS ISCSI_TCP"
```
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: Kernel checks should be non-fatal, using ~SCSI_ISCSI_ATTRS and similar. The kernel config isn't always properly detected during `build_image` causing things to fail erroneously.

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	sys-fs/lsscsi
+	sys-apps/util-linux"
+
+S="${WORKDIR}/${MY_PV}"
+
+pkg_setup() {
+	linux-info_pkg_setup
+
+	if kernel_is -lt 2 6 16; then
+		die "Sorry, your kernel must be 2.6.16-rc5 or newer!"
```
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: ewarn instead

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild (hunk 2 above)
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: oh, nm. missed the line below, these are already done as warnings..

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+	# SSL (--with-security) is broken
+	econf $(use_with slp) \
+		--without-security
+}
+
+src_compile() {
+	use debug && append-flags -DDEBUG_TCP -DDEBUG_SCSI
+
+	KSRC="${KV_DIR}" CFLAGS="" \
```
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: What does the build need the kernel source for?

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+	# udev pieces
+	insinto /usr/lib/udev/rules.d
+	doins "${FILESDIR}"/99-iscsi.rules
+	insopts -m0755
+	insinto /usr/lib/udev/scripts
+	doins "${FILESDIR}"/iscsidev.sh
+	insopts -m0644
+
+	keepdir /var/db/iscsi
```
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: needs a tmpfiles rule to create this on boot if missing.

File: sys-block/open-iscsi/open-iscsi-2.0.873.ebuild [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	fperms 700 /var/db/iscsi
+	fperms 600 /etc/iscsi/iscsid.conf
```
Comment (marineam) [inline sys-block/open-iscsi/open-iscsi-2.0.873.ebuild]: This is mostly an example file so installing to /etc is fine. Unless of course things just explode if it is missing, in which case we need a tmpfiles rule to make sure it exists.

**Comment by mischief** [comment]
@marineam i think all the things you mentioned are fixed. PTAL

upstream, there are systemd units, but they would need to be slightly modified for our use.

**Comment by mischief** [comment]
@marineam ping. should we go to upstream master for the systemd units, e.g. https://github.com/mikechristie/open-iscsi/tree/master/etc/systemd ?

**Comment by marineam** [comment]
@mischief yes, lets use those units as a starting place. May need to check if the socket unit works at all with the current release though.

**Comment by marineam** [comment]
@mischief I remain skeptical of `iscsidev.sh` but we can merge this as-is and improve with follow up PRs.

**Comment by dlaube** [comment]
echoing @darrensessions here as well. Will multipathd be included? 
iSCSI is nice, but HA iSCSI is nicer :)

**Comment by marineam** [comment]
@dlaube we'll see, we have yet to actually try out this stuff. But probably

**Pull request merged**